Always copy pointdata to the output, without additional weighting and interpolation.

--- a/Filters/Core/vtkQuadricDecimation.cxx	2024-05-27 12:49:51.834643400 +0800
+++ b/Filters/Core/vtkQuadricDecimation.cxx	2024-05-27 12:53:54.287779200 +0800
@@ -63,6 +63,7 @@
 
   this->AttributeErrorMetric = 0;
   this->VolumePreservation = 0;
+  this->AlwaysCopyPointData = 0;
   this->ScalarsAttribute = 1;
   this->VectorsAttribute = 1;
   this->NormalsAttribute = 1;
@@ -281,7 +282,7 @@
   polys->DeepCopy(input->GetPolys());
   this->Mesh->SetPolys(polys);
   polys->Delete();
-  if (this->AttributeErrorMetric || this->MapPointData)
+  if (this->AttributeErrorMetric || this->MapPointData || this->AlwaysCopyPointData)
   {
     this->Mesh->GetPointData()->DeepCopy(input->GetPointData());
   }
--- a/Filters/Core/vtkQuadricDecimation.h	2024-05-27 12:49:51.834643400 +0800
+++ b/Filters/Core/vtkQuadricDecimation.h	2024-05-27 12:51:59.970096500 +0800
@@ -146,6 +146,17 @@
 
   ///@{
   /**
+   * If on, point data will always be copied to the output, regardless of
+   * whether or not AttributeErrorMetric is active. By default
+   * AlwaysCopyPointData is off.
+   */
+  vtkSetMacro(AlwaysCopyPointData, vtkTypeBool);
+  vtkGetMacro(AlwaysCopyPointData, vtkTypeBool);
+  vtkBooleanMacro(AlwaysCopyPointData, vtkTypeBool);
+  ///@}
+
+  ///@{
+  /**
    * If attribute errors are to be included in the metric (i.e.,
    * AttributeErrorMetric is on), then the following flags control which
    * attributes are to be included in the error calculation. By default all
@@ -288,6 +299,7 @@
   int OutputPointsPrecision;
   vtkTypeBool AttributeErrorMetric;
   vtkTypeBool VolumePreservation;
+  vtkTypeBool AlwaysCopyPointData;
 
   bool MapPointData = false;
 
